AVFoundation-Combine/Video Player/VideoPlayerViewController.swift

```diff
@@ -0,0 +1,163 @@
+    }
+    
+    // MARK: Video Player setup
+    
+    private func setup(_ player: AVPlayer) {
+        player.currentItemPublisher()
+            .compactMap { $0 }
+            .sink {[weak self] item in
+                self?.setup(item)
+                self?.avPlayerViewController.player?.play()
+            }
+            .store(in: &subscriptions)
+        
+        player.playheadProgressPublisher()
+            .filter {progress in
+                !self.isProgressSliderScrubbing
+            }
+            .map { Float($0) }
+            .assign(to: \.value, on: customUI.progressSlider)
+            .store(in: &subscriptions)
+        
+        let rateStream = player.ratePublisher().share()
+        
+        rateStream.receive(on: DispatchQueue.main)
+            .map { $0 == 0.0 ? "Play" : "Pause" }
+            .assign(to: \.accessibilityLabel, on: customUI.playbackButton)
+            .store(in: &subscriptions)
+        
+        rateStream.receive(on: DispatchQueue.main)
+            .map { $0 == 0.0 ? UIImage(named: "Play") : UIImage(named: "Pause") }
+            .sink {[weak self] image in
+                self?.customUI.playbackButton.setImage(image, for: .normal)
+            }
+            .store(in: &subscriptions)
+        
+        avPlayerViewController.player = player
+        player.replaceCurrentItem(with: AVPlayerItem(url: videoURL))
+    }
+    
+    private func setup(_ item: AVPlayerItem) {
+        
+        item.isPlaybackLikelyToKeepUpPublisher()
+            .receive(on: DispatchQueue.main)
+            .assign(to: \.isHidden, on: customUI.loadingIndicator)
+            .store(in: &subscriptions)
+        
+        item.isPlaybackBufferEmptyPublisher()
+            .receive(on: DispatchQueue.main)
+            .map { !$0 }
+            .assign(to: \.isHidden, on: customUI.loadingIndicator)
+            .store(in: &subscriptions)
+        
+        let statusStream = item.statusPublisher().share()
+        
+        statusStream.receive(on: DispatchQueue.main)
+            .map { $0 == .readyToPlay }
+            .assign(to: \.isEnabled, on: customUI.playbackButton)
+            .store(in: &subscriptions)
+        
+        statusStream.receive(on: DispatchQueue.main)
+            .map { $0 == .readyToPlay }
+            .assign(to: \.isEnabled, on: customUI.progressSlider)
+            .store(in: &subscriptions)
+        
+        statusStream.receive(on: DispatchQueue.main)
+            .map { $0 == .readyToPlay ? 1.0 : 0.25 }
+            .assign(to: \.alpha, on: customUI.playbackButton)
+            .store(in: &subscriptions)
+        
+        statusStream.receive(on: DispatchQueue.main)
+            .map { $0 == .readyToPlay ? 0.5 : 1.0 }
+            .assign(to: \.alpha, on: customUI.logoImageView)
+            .store(in: &subscriptions)
+        
+        item.durationPublisher()
```

Comment:
I feel like this could use a `removeDuplicates` operator to only communicate distinct changes.